sim-lib/src/lib.rs

```diff
@@ -275,13 +277,48 @@ impl Simulation {
         Ok(())
     }
 
+    // validate_node_network validates that the nodes that are on the same bitcoin network.
+    // e.g. node A and B are on testnet, node C is on mainnet should return an error.
+    async fn validate_node_network(&self) -> Result<(), LightningError> {
+        let mut valid_network = Option::None;
+        let mainnet = Network::Bitcoin.to_string().to_lowercase();
```

Comment:
This string is "bitcoin" which won't match the `network` set by LND "mainnet" (there's divergence between LDK and LND in naming).

Reply:
Yes, an oversight on my part. fixed here -> https://github.com/bitcoin-dev-project/sim-ln/pull/112/commits/5d2bfac7f1a9d20630899b10602bdcf8cae38b75